pkg/eval/ievaluator.go

```diff
@@ -4,13 +4,27 @@ import (
 	"google.golang.org/protobuf/types/known/structpb"
 )
 
+type StateChangeNotificationType string
+
+const (
+	NotificationDelete StateChangeNotificationType = "delete"
+	NotificationCreate StateChangeNotificationType = "write"
+	NotificationUpdate StateChangeNotificationType = "update"
+)
+
+type StateChangeNotification struct {
+	Type    StateChangeNotificationType `json:"type"`
+	Source  string                      `json:"source"`
+	FlagKey string                      `json:"flagKey"`
+}
+
 /*
 IEvaluator implementations store the state of the flags,
 do parsing and validation of the flag state and evaluate flags in response to handlers.
 */
 type IEvaluator interface {
 	GetState() (string, error)
-	SetState(source string, state string) error
+	SetState(source string, state string) ([]StateChangeNotification, error)
```

Comment:
This interface is becoming ugly, maybe we should pass an object to encapsulate all this?

```
SetState(state IState) 
```

Reply:
I think matching the types may lead to a little confusion, we could wrap the input arguments as 
```go
type IEvaluatorStateConfiguration struct {
	Source string
	State  string
}
```
and update the output argument type to be `[]IEvaluatorStateChange`

`SetState(state IEvaluatorStateConfiguration) ([]IEvaluatorStateChange, error)`

what do you think @AlexsJones

Reply:
Works for me yes